src/mlir/consumer.cc

```diff
@@ -258,12 +258,12 @@ namespace mlir::verona
     /// Closes the function, checking the return value
     void post(Function& node)
     {
+      // Automatically clean up
+      ScopeCleanup([&]() { symbolTable().popScope(); });
```

Comment:
Does this actually work?  I was under the impression that this would create a new object that has the lifetime of the expression and so does the pop immediately.  I'd expect this to be more like:

```
ScopeCleanup pop([&]() { symbolTable().popScope(); });
```
I wrote a simple test that confirmed my suspicion:

```c++
ScopeCleanup([]() { foo(); });
baz();
```

Generates:

```llvm
  tail call void @_Z3foov() #2
  tail call void @_Z3bazv() #2
```

```suggestion
      ScopeCleanup pop([&]() { symbolTable().popScope(); });
```

Reply:
D'oh!

Reply:
I think I have tests that check the stack not being popped, but I don't have one for it being popped too early.